programs/vote/benches/cu_vote.rs

```diff
@@ -0,0 +1,165 @@
+    solana_clock::Clock,
+    solana_instruction::{error::InstructionError, AccountMeta},
+    solana_program_runtime::invoke_context::mock_process_instruction,
+    solana_pubkey::Pubkey,
+    solana_rent::Rent,
+    solana_sdk::sysvar,
+    solana_sdk_ids::vote::id,
+    solana_vote_program::{
+        vote_instruction::VoteInstruction,
+        vote_processor::Entrypoint,
+        vote_state::{create_account, VoteAuthorize, VoteInit, VoteState},
+    },
+};
+
+fn create_default_rent_account() -> AccountSharedData {
+    account::create_account_shared_data_for_test(&Rent::free())
+}
+
+fn create_default_clock_account() -> AccountSharedData {
+    account::create_account_shared_data_for_test(&Clock::default())
+}
+
+fn create_test_account() -> (Pubkey, AccountSharedData) {
+    let rent = Rent::default();
+    let balance = VoteState::get_rent_exempt_reserve(&rent);
+    let vote_pubkey = solana_pubkey::new_rand();
+    (
+        vote_pubkey,
+        create_account(&vote_pubkey, &solana_pubkey::new_rand(), 0, balance),
+    )
+}
+
+struct TestSetup {}
+
+impl TestSetup {
+    fn new() -> TestSetup {
+        Self {}
+    }
+
+    fn process_instruction(
+        instruction_data: &[u8],
+        transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
+        instruction_accounts: Vec<AccountMeta>,
+        expected_result: Result<(), InstructionError>,
+    ) -> Vec<AccountSharedData> {
+        mock_process_instruction(
+            &id(),
+            Vec::new(),
+            instruction_data,
+            transaction_accounts,
+            instruction_accounts,
+            expected_result,
+            Entrypoint::vm,
+            |_invoke_context| {},
+            |_invoke_context| {},
+        )
+    }
+    pub fn test_initialize_account(&self) {
+        let vote_pubkey = solana_pubkey::new_rand();
+        let vote_account = AccountSharedData::new(100, VoteState::size_of(), &id());
+        let node_pubkey = solana_pubkey::new_rand();
+        let node_account = AccountSharedData::default();
+        let instruction_data = serialize(&VoteInstruction::InitializeAccount(VoteInit {
+            node_pubkey,
+            authorized_voter: vote_pubkey,
+            authorized_withdrawer: vote_pubkey,
+            commission: 0,
+        }))
+        .unwrap();
+        let instruction_accounts = vec![
+            AccountMeta {
+                pubkey: vote_pubkey,
+                is_signer: false,
+                is_writable: true,
+            },
+            AccountMeta {
+                pubkey: sysvar::rent::id(),
+                is_signer: false,
+                is_writable: false,
+            },
+            AccountMeta {
+                pubkey: sysvar::clock::id(),
+                is_signer: false,
+                is_writable: false,
+            },
+            AccountMeta {
+                pubkey: node_pubkey,
+                is_signer: true,
+                is_writable: false,
+            },
+        ];
+
```

Comment:
this block is executed for each bench loop, that'd distort the measure of `mock_process_instruction()`, could move them to `TestSetup::new()`.